gateway/proxy_muxer.go

```diff
@@ -96,9 +97,13 @@ func (h *handleWrapper) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if NewRelicApplication != nil {
-		txn := NewRelicApplication.StartTransaction(r.URL.Path, w, r)
+		txn := NewRelicApplication.StartTransaction(r.URL.Path)
+		w = txn.SetWebResponse(w)
+
+		newrelic.Context.Set(r, txn)
+
 		defer txn.End()
-		h.router.ServeHTTP(txn, r)
+		h.router.ServeHTTP(w, r)
```

Comment:
@jeffy-mathew track the other issue here:

> attaching nrgorilla middleware not used

https://github.com/newrelic/go-agent/blob/v3/integrations/nrgorilla/v1.2.2/v3/integrations/nrgorilla/nrgorilla.go#L113